packages/block-library/src/columns/edit.native.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+import { dropRight, times } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	PanelBody,
+	StepperControl,
+	Toolbar,
+	ToolbarButton,
+} from '@wordpress/components';
+import {
+	InspectorControls,
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+} from '@wordpress/block-editor';
+import { withDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { withViewportMatch } from '@wordpress/viewport';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import Icon from './icon';
+
+/**
+ * Allowed blocks constant is passed to InnerBlocks precisely as specified here.
+ * The contents of the array should never change.
+ * The array should contain the name of each block that is allowed.
+ * In columns block, the only block we allow is 'core/column'.
+ *
+ * @constant
+ * @type {string[]}
+ */
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+/**
+ * Number of columns to assume for template in case the user opts to skip
+ * template option selection.
+ *
+ * @type {number}
+ */
+const DEFAULT_COLUMNS = 2;
+const MIN_COLUMNS_NUMBER = 2;
+const MAX_COLUMNS_NUMBER = 6;
+
+function ColumnsEditContainer( {
+	attributes,
+	updateBlockSettings,
+	blockListSettings,
+	updateAlignment,
+	updateColumns,
+	clientId,
+	isMobile,
+} ) {
+	const { verticalAlignment } = attributes;
+	const { width } = blockListSettings;
+
+	const { columnCount } = useSelect( ( select ) => {
+		return {
+			columnCount: select( 'core/block-editor' ).getBlockCount(
+				clientId
+			),
+		};
+	} );
+
+	useEffect( () => {
+		updateColumns(
+			columnCount,
+			Math.min( MAX_COLUMNS_NUMBER, columnCount || DEFAULT_COLUMNS )
+		);
+	}, [] );
+
+	const getColumnsInRow = ( containerWidth, columnsNumber ) => {
```

Comment:
It's the logic which will be replaced with Luke's Hook which will get container dimensions size